helpers/pathspec.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2016-present The Hugo Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package helpers
+
+// PathSpec holds methods that decides how paths in URLs and files in Hugo should look like.
+type PathSpec struct {
+	disablePathToLower             bool
+	removePathAccents              bool
+	defaultContentLanguageInSubdir bool
+	uglyURLs                       bool
+	canonifyURLs                   bool
+
+	// language
+	multilingual           bool
+	defaultContentLanguage string
+	currentContentLanguage *Language
+
+	// pagination path handling
+	paginatePath string
+}
+
+func NewPathSpecFromConfig(config ConfigProvider) PathSpec {
+	// TODO(bep) check global vs language scope
+	return PathSpec{
+		disablePathToLower:             config.GetBool("DisablePathToLower"),
```

Comment:
> It looks like everything you're loading is global. 

Right now, everything Is loaded language first. Misused is maybe a wrong word (the user is shooting him- or herself in the foot), but we are probably making promises we cannot keep.